src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.NameGenerator.cs

```diff
@@ -19,7 +19,7 @@ internal class NameGenerator
         {
             internal static ImmutableArray<Words> GetBaseNames(ITypeSymbol type, bool pluralize)
             {
-                var baseName = TryRemoveInterfacePrefix(type);
+                var baseName = TryRemoveKnownPrefixes(type, pluralize);
```

Comment:
name parameter for pluralize as you're naming them differently with slightly different semantics.  THe way this code looks like, it looks like if you had IFoo and were calling with pluralize=true, you would get `foos`, when you don't actually get taht.